### PR TITLE
Update README with examples and links to feedback collection

### PR DESCRIPTION
**Pull request by stasm**
I'd like to improve the README so that:

* It tells a better story of what MessageFormat is.
* It showcases a few examples of the draft syntax.
* It points to the existing tech preview implementations.
* It redirects users who want to share feedback to appropriate forums.

### CODE REVIEW

**Review by romulocintra — APPROVED** [review]
Thank you @stasm  for this PR

**Review by echeran — CHANGES_REQUESTED** [review]
Flipping the review bit so that we don't merge until Mark's comment is addressed, at least.  We shouldn't knowingly be precise on things that are in reality TBD / done differently.

File: README.md [hunk 1]
```diff
@@ -1,20 +1,76 @@
...
-
-## Guidelines
+The Message Format Working Group (MFWG) is tasked with developing an industry standard for the representation of localizable message strings to be a successor to [ICU MessageFormat](https://unicode-org.github.io/icu/userguide/format_parse/messages/). MFWG will recommend how to remove redundancies, make the syntax more usable, and support more complex features, such as gender, inflections, and speech. MFWG will also consider the integration of the new standard with programming environments, including, but not limited to, ICU, DOM, and ECMAScript, and with localization platform interchange. The output of MFWG will be a specification for the new syntax.
```
Comment (eemeli) [inline README.md:7]: Not necessarily requiring action, but seeing what's different here would've been easier if this file used [semantic line breaks](https://sembr.org/).

**Review by eemeli — APPROVED** [review]

**Review by echeran — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,20 +1,76 @@
...
 
+- [Why ICU MessageFormat Needs a Successor](docs/why_mf_next.md)
 - [Goals and Non-Goals](guidelines/goals.md)
+- [Record of Consensus Decisions](docs/consensus_decisions.md)
+
+## MessageFormat 2.0 Draft Syntax
```
Comment (echeran) [inline README.md:13]:
```suggestion
## MessageFormat 2 Draft Syntax
```

File: README.md [hunk 3]
```diff
@@ -1,20 +1,76 @@
...
+    when * 0 {{$hostName} does not give a party.}
+    when * 1 {{$hostName} invites {$guestName} to their party.}
+    when * 2 {{$hostName} invites {$guestName} and one other person to their party.}
+    when * * {{$hostName} invites {$guestName} and {$guestsOther} other people to their party.}
+
+See more examples and the formal definition of the grammar in [spec/syntax.md](./spec/syntax.md).
+
+### Implementations
+
+* Java: [`com.ibm.icu.message2`](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4j/index.html?com/ibm/icu/message2/package-summary.html), part of ICU 72 released in October 2022, is a _tech preview_ implementation of the MessageFormat 2.0 syntax, together with a formatting API.
```
Comment (echeran) [inline README.md:58]:
```suggestion
* Java: [`com.ibm.icu.message2`](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4j/index.html?com/ibm/icu/message2/package-summary.html), part of ICU 72 released in October 2022, is a _tech preview_ implementation of the MessageFormat 2 syntax, together with a formatting API.
```

File: README.md [hunk 4]
```diff
@@ -1,20 +1,76 @@
...
+* JavaScript: [`messageformat`](https://github.com/messageformat/messageformat/tree/master/packages/mf2-messageformat) 4.0 implements the MessageFormat 2.0 syntax, together with a polyfill of the runtime API proposed for ECMA-402.
```
Comment (echeran) [inline README.md:59]:
```suggestion
* JavaScript: [`messageformat`](https://github.com/messageformat/messageformat/tree/master/packages/mf2-messageformat) 4.0 implements the MessageFormat 2 syntax, together with a polyfill of the runtime API proposed for ECMA-402.
```

**Review by echeran — COMMENTED** [review]
Based on offline discussion with @stasm and @eemeli, I'm suggesting changing the "2.0" to "2" so that we don't give the impression that our current working specification has the formalism or completion that an unqualified semver like "2.0" implies.  Especially when measuring progress in relation to the CLDR-TC guidance on the scope of our goals for what 2.0 entails.

optional: the specification's docs on [syntax](https://github.com/unicode-org/message-format-wg/blob/main/spec/syntax.md) and [formatting](https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md) both include "DRAFT" alongside the title, but if you agree with these changes, then it would be good to update those instances of "2.0" to "2", too.

**Comment by stasm** [comment]
> optional: the specification's docs on [syntax](https://github.com/unicode-org/message-format-wg/blob/main/spec/syntax.md) and [formatting](https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md) both include "DRAFT" alongside the title, but if you agree with these changes, then it would be good to update those instances of "2.0" to "2", too.

I agree with the direction of this change across the repo, but I think we should do it in a separate PR.

**Comment by echeran** [comment]
> I agree with the direction of this change across the repo, but I think we should do it in a separate PR.

Sounds good.

**Pull request merged**